src/app/pages/Swap/modals/SwapSettings/index.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { memo, useCallback, useState } from 'react';
+
+import { isEmpty } from 'lodash';
+import { Controller, SubmitHandler, useForm } from 'react-hook-form-v7';
+
+import AssetField from 'app/atoms/AssetField';
+import { ActionsButtonsBox, PageModal } from 'app/atoms/PageModal';
+import { StyledButton } from 'app/atoms/StyledButton';
+import { ReactComponent as InfoIcon } from 'app/icons/info-icon.svg';
+import { T } from 'lib/i18n';
+
+export type Inputs = {
+  slippageTolerance: number | undefined;
+};
+
+interface SelectTokenModalProps {
+  onSubmit: SubmitHandler<Inputs>;
+  opened: boolean;
+  onRequestClose: EmptyFn;
+}
+
+const SwapSettingsModal = memo<SelectTokenModalProps>(({ onSubmit, opened, onRequestClose }) => {
+  const options = [0.5, 1, 'Custom'] as const;
+  const [selected, setSelected] = useState<(typeof options)[number]>(0.5);
+
+  const {
+    watch,
+    control,
+    handleSubmit,
+    setValue,
+    formState: { errors, submitCount, isSubmitting }
+  } = useForm<Inputs>();
+  const formSubmitted = submitCount > 0;
+
+  const slippageValue = watch('slippageTolerance');
+
+  const handleInputClean = useCallback(
+    () => setValue('slippageTolerance', undefined, { shouldValidate: formSubmitted }),
+    [setValue, formSubmitted]
+  );
+
+  const handleOptionChange = (option: (typeof options)[number]) => {
+    setSelected(option);
+    if (option === 'Custom') {
+      setValue('slippageTolerance', undefined, { shouldValidate: formSubmitted });
+    } else if (option === 0.5 || option === 1) {
+      setValue('slippageTolerance', option, { shouldValidate: formSubmitted });
+    }
+  };
+
+  const handleApplyClick = () => {
+    if (selected !== 'Custom') {
+      setValue('slippageTolerance', selected, { shouldValidate: formSubmitted });
+    }
+    handleSubmit(onSubmit)();
+  };
+
+  return (
+    <PageModal title="Swap Settings" opened={opened} onRequestClose={onRequestClose}>
+      <div className="mt-5 px-4">
+        <span className="font-semibold text-xs">
+          <T id="slippageTolerance" />
+        </span>
+
+        <div className="bg-secondary-low rounded-6 p-3 flex gap-1 items-start mt-2">
+          <InfoIcon className="min-w-5 min-h-5 mt-0.5" />
+          <span className="text-xs">
+            The swap will fail if the price changes unfavorably by more than this percentage. Too high value may lead to
+            an unfavorable trade.
+          </span>
+        </div>
```

Comment:
use `CaptionAlert` component instead, also pls add new text to `en/message.json`